### PR TITLE
perf: remove repeated pool exists check

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Removing repeated defense-in-depth sanity check. We should be checking pool existence much earlier in the execution flow.

Old:

image

New:

image

With 100k positions and 180k ticks traversed:
```
go test -bench . -run BenchmarkSwapExactAmountIn -timeout=30m -v -count=3
goos: linux
goarch: amd64
pkg: github.com/osmosis-labs/osmosis/v15/x/concentrated-liquidity
cpu: Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz
BenchmarkSwapExactAmountIn
BenchmarkSwapExactAmountIn-4                           1        2412393389 ns/op
BenchmarkSwapExactAmountIn-4                           1        2406438654 ns/op
BenchmarkSwapExactAmountIn-4                           1        2307147354 ns/op
```

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [x] N/A

### CODE REVIEW

**Comment by ValarDragon** [comment]
So this also wouldn't be expensive if we weren't unmarshalling the pool, and just checking that there is a value with non-zero length at that key. There should be helpers in `osmoutils` for this

**Comment by ValarDragon** [comment]
This PR isn't including IAVL time AFAICT, as there is no commit happening

**Comment by p0mvn** [comment]
> This PR isn't including IAVL time AFAICT, as there is no commit happening

Merged https://github.com/osmosis-labs/osmosis/pull/5282 here.

The original profiles and benchmarks include IAVL time but not level db compactions as they were done prior to the commit that removed `Commit()`: [1a94f36](https://github.com/osmosis-labs/osmosis/pull/5278/commits/1a94f3683e769363b5c1e32d9134c8f1a83927a3)

I will re-bench with level db for the next improvement. From a single run with level db, it ends up being approximately the same

**Comment by p0mvn** [comment]
> So this also wouldn't be expensive if we weren't unmarshalling the pool, and just checking that there is a value with non-zero length at that key. There should be helpers in `osmoutils` for this

Agreedbut I don't think there is much value in this check. Pool existence should be checked at a higher level of abstraction.

I can implement the suggestion if people are feeling strongly but removing it feels cleaner to me.

File: x/concentrated-liquidity/bench_test.go [hunk 1]
```diff
@@ -198,7 +200,9 @@ func runBenchmark(b *testing.B, testFunc func(b *testing.B, s *BenchTestSuite, p
 
 		swapAmountIn := sdk.MustNewDecFromStr(amountIn).TruncateInt()
 		largeSwapInCoin := sdk.NewCoin(denomIn, swapAmountIn)
+		// Commit so that the changes are propagated to IAVL.
 		s.Commit()
+		noError(b, err)
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/bench_test.go:205]: What error is this checking?

**Review by czarcas7ic — APPROVED** [review]
I don't think the extra noError call is needed, other than that LGTM

**Comment by p0mvn** [comment]
> LGTM, I wonder if this function should be made private
> 
> (Lets comment that its up to the caller to ensure the poolID exists?)

Added comment. Looks like it has to be exported for some of the superfluid test that rely on this function.

I don't think there are any security concerns. The worst-case scenario if the pool does not exist is that the tick will not be found

File: x/concentrated-liquidity/bench_test.go (hunk 1 above)
Comment (p0mvn) [reply x/concentrated-liquidity/bench_test.go]: Removed

**Pull request merged**